source/state_representation/include/state_representation/State.hpp

```diff
@@ -159,6 +159,12 @@ class State : public std::enable_shared_from_this<State> {
    */
   void set_empty(bool empty = true);
 
+  /**
+   * @brief Print the state
+   * @param state_type The type of the state to print
+   */
+  virtual std::string print_state(const StateType& state_type) const;
```

Comment:
Yeah that was a mistake. I guess I thought that if I call the parent `to_string`, it would take the type of the parent but for that operation, there is no new object created so everything works fine without that. Thanks